comixed-webui/src/app/collections/pages/collection-detail/collection-detail.component.ts

```diff
@@ -108,6 +118,26 @@ export class CollectionDetailComponent implements OnInit, OnDestroy {
     });
     this.queryParamsSubscribe = this.activatedRoute.queryParams.subscribe(
       params => {
+        if (!!params[QUERY_PARAM_ARCHIVE_TYPE]) {
+          const archiveType = params[QUERY_PARAM_ARCHIVE_TYPE];
+          this.logger.debug('Received archive type query param:', archiveType);
+          switch (archiveType) {
+            case ArchiveType.CBZ:
+              this.archiveTypeFilter = ArchiveType.CBZ;
+              break;
+            case ArchiveType.CBR:
+              this.archiveTypeFilter = ArchiveType.CBR;
+              break;
+            case ArchiveType.CB7:
+              this.archiveTypeFilter = ArchiveType.CB7;
+              break;
+            default:
+              this.archiveTypeFilter = null;
+          }
+        } else {
+          this.logger.debug('Resetting archive type filter');
+          this.archiveTypeFilter = null;
+        }
```

Comment:
I wonder if something like this could be more concise and reduce the number of duplicated lines (but i'm not sure that the performance are not altered) :
          if (new Set([ArchiveType.CBZ, ArchiveType.CBR, ArchiveType.CB7]).has(archiveType)) {
            this.archiveTypeFilter = archiveType;
          } else {
            this.archiveTypeFilter = null;
          }

Reply:
I think your suggestion is a good one. I'll implement it tonight and update the PR.

Reply:
Pushed the updated code. I removed that code, plus the same check in another spot, since there was an existing method to look for the archive type from a string.